packages/stark-ui/src/modules/keyboard-directives/directives/transform-input.directive.spec.ts

```diff
@@ -0,0 +1,111 @@
+import { Component, OnInit } from "@angular/core";
+import { ComponentFixture, TestBed } from "@angular/core/testing";
+import { FormControl, FormsModule, ReactiveFormsModule } from "@angular/forms";
+import { By } from "@angular/platform-browser";
+import { MockStarkLoggingService } from "@nationalbankbelgium/stark-core/testing";
+import { STARK_LOGGING_SERVICE } from "@nationalbankbelgium/stark-core";
+import { StarkTransformInputDirective } from "./transform-input.directive";
+
+/**
+ * @ignore
+ * Mocks an InputEvent on the element with the given value.
+ * @param value - the value to set the element to.
+ * @param element - the HTMLInputElement to mock the event on
+ */
+const mockInputEvent: (value: string, element: HTMLInputElement) => void = (value: string, element: HTMLInputElement): void => {
+	const event: Event = document.createEvent("Event");
+	event.initEvent("input", true, true);
+	element.value = value;
+	element.dispatchEvent(event);
+};
+
+describe("TransformsInputDirective with ngModel", () => {
+	/**
+	 * @ignore
+	 * mocked wrapper for the directive
+	 */
+	@Component({
+		selector: "test-component",
+		template: "<input [(ngModel)]='value' starkTransformInput='upper'/>"
+	})
+	class TestComponent {
+		public value: string = "";
+	}
+
+	let fixture: ComponentFixture<TestComponent>;
+
+	beforeEach(() => {
+		TestBed.configureTestingModule({
+			declarations: [StarkTransformInputDirective, TestComponent],
+			imports: [FormsModule],
+			providers: [{ provide: STARK_LOGGING_SERVICE, useValue: new MockStarkLoggingService() }]
+		});
+
+		fixture = TestBed.createComponent(TestComponent);
+		// trigger initial data binding
+		fixture.detectChanges();
+	});
+
+	it("should set the value to uppercase", () => {
+		expect(fixture.componentInstance.value).toBe("", "field 'value' should start as empty string ");
+
+		const inputElement: HTMLInputElement = <HTMLInputElement>fixture.debugElement.query(By.css("input")).nativeElement;
+		mockInputEvent("a", inputElement);
```

Comment:
You can also mock typing a non-word character and also an uppercase char to test that the chars remains "as is'... you know... just to be sure ;)